ftw/trash/browser/trash.py

```diff
@@ -98,14 +99,18 @@ def clean_trash(self, REQUEST):
 
             paths_to_delete = filter_children_in_paths(
                 [brain.getPath() for brain in catalog(query)])
-            for path in paths_to_delete:
-                obj = self.context.restrictedTraverse(path)
-                got_path = '/'.join(obj.getPhysicalPath())
-                if got_path != path:
-                    raise ValueError('Unexpectly found path {!r} when looking for {!r}'.format(
-                        got_path, path))
-
-                aq_parent(aq_inner(obj)).manage_immediatelyDeleteObjects([obj.getId()])
+
+            # The user maybe able to clean the trash, but depending on the workflow the user may not have
+            # delete permission in the given context
+            with api.env.adopt_roles(['Manager']):
```

Comment:
We've had a problem recently where this did not what we've expected. It seems to work here, but if we run into problems we should replace it with something like [GEVER's `elevated_privileges`](https://github.com/4teamwork/opengever.core/blob/e2414979bc679aedf87b4bbc7dc376842b090362/opengever/base/security.py#L24).